arelle/plugin/validate/ESEF/Dimensions.py

```diff
@@ -31,87 +31,87 @@ def addDomMbrs(sourceDomMbr, ELR, membersSet):
-                            elrPrimaryItems["*"].add(hcPrimaryItem) # members of any ELR
-                hcPrimaryItems.clear()
-                hcMembers.clear()
-                                 
-    # find primary items with other dimensions in 
-    #for ELR, priItems in elrPrimaryItems.items():
-    #    if ELR != LineItemsNotQualifiedLinkrole:
-    #        # consider any pri item in not reported non-dimensionally
-    #        i = set(hcPrimaryItem
-    #                for hcPrimaryItem in (priItems & elrPrimaryItems.get(LineItemsNotQualifiedLinkrole, set()))
-    #                if not any(not f.context.qnameDims for f in val.modelXbrl.factsByQname.get(hcPrimaryItem.qname,())))
-    #        if i:
-    #            val.modelXbrl.warning("ESEF.3.4.2.extensionTaxonomyLineItemIncorrectlyLinkedToNonDimensionallyQualifiedHypercube",
-    #                _("Dimensional line item SHOULD NOT also be linked to \"not dimensionally qualified\" hypercube from %(linkrole)s, primary item %(qnames)s"),
-    #                modelObject=i, linkrole=ELR, qnames=", ".join(sorted(str(c.qname) for c in i)))
-
-    # reported pri items not in LineItemsNotQualifiedLinkrole
-    i = set(concept
-            for qn, facts in val.modelXbrl.factsByQname.items()
-            if any(not f.context.qnameDims for f in facts if f.context is not None)
-            for concept in (val.modelXbrl.qnameConcepts.get(qn),)
-            if concept is not None and 
-               concept not in elrPrimaryItems.get(LineItemsNotQualifiedLinkrole, set()) and
-               concept not in elrPrimaryItems.get("*", set()))
-    if i:
-        val.modelXbrl.warning("ESEF.3.4.2.extensionTaxonomyLineItemNotLinkedToAnyHypercube",
-            _("Dimensional line item reported non-dimensionally SHOULD be linked to \"not dimensionally qualified\" hypercube %(linkrole)s, primary item %(qnames)s"),
-            modelObject=i, linkrole=LineItemsNotQualifiedLinkrole, qnames=", ".join(sorted(str(c.qname) for c in i)))
-    # pri items in LineItemsNotQualifiedLinkrole which are not used in report non-dimensionally
-    # check no longer in Filer Manual as of 2021
-    #i = set(hcPrimaryItem
-    #       for hcPrimaryItem in elrPrimaryItems.get(LineItemsNotQualifiedLinkrole, set())
-    #       if not any(not f.context.qnameDims 
-    #                  for f in val.modelXbrl.factsByQname.get(hcPrimaryItem.qname,())
-    #                  if f.context is not None))
-    #if i:
-    #    val.modelXbrl.warning("ESEF.3.4.2.extensionTaxonomyLineItemIncorrectlyLinkedToNonDimensionallyQualifiedHypercube",
-    #        _("Dimensional line item not reported non-dimensionally has no need to be linked to \"not dimensionally qualified\" hypercube %(linkrole)s, primary item %(qnames)s"),
-    #        modelObject=i, linkrole=LineItemsNotQualifiedLinkrole, qnames=", ".join(sorted(str(c.qname) for c in i)))
+        # reported pri items not in LineItemsNotQualifiedLinkrole
+        i = set(concept
```

Comment:
Pre-existing: I _think_ this means `unlinkedPrimaryItems`?  might be cool to update the name
Pre-existing: Pretty complicated list comprehension, might increase readability to break into a few statements